serveradmin/servershell/static/js/servershell/result.js

```diff
@@ -244,14 +244,26 @@ register_inline_editing = function(cell) {
             let attribute = servershell.get_attribute(attribute_id);
 
             // When the user types 'false' use empty string so that it casts to false
-            if (attribute.type === 'boolean' && (value === 'false' || value === '0'))
+            if (attribute.type === 'boolean' && (value === 'false' || value === '0')) {
                 value = '';
+            }
 
             if (value === '') {
                 servershell.delete_attribute(object_id, attribute_id)
             }
             else {
                 if (multi) {
+                    // Cast to matching type before comparing
+                    value = value.map(function(v) {
+                       if (attribute.type === 'number') {
```

Comment:
Above it looks like a multi value can just have values of one attribute type. So, multiple numbers or multiple booleans or the like. Now it looks like a multi value can have values of different attribute types.
What is correct? As I would suggest to rather use one way of handling it, if this is possible. Then you can easily modulize it later.

Reply:
A multi attribute can only have values of one type. I just wrote it this way because it is shorter.

Reply:
Then I would suggest to use one way, to easier modularize it later. The way in this line might be a bit slower as you are doing not needed checks for the attribute type for every value, even though the attribute type can't be different.

Reply:
Modularisation is not planned in the near future. Since we only put in small parts of texts (maybe max 256 characters) I would consider that a micro optimisation.

I will adjust it.